library/common/network/proxy_settings.h

```diff
@@ -71,14 +78,14 @@ struct ProxySettings {
       return address_->asString();
     }
     if (!hostname_.empty()) {
-      return hostname_;
+      return absl::StrCat(hostname_, ":", port_);
     }
     return "no_proxy_configured";
   }
 
   bool operator==(ProxySettings const& rhs) const {
     if (this->address() == nullptr || rhs.address() == nullptr) {
-      return this->address() == nullptr && rhs.address() == nullptr;
+      return this->hostname() == rhs.hostname() && this->port() == rhs.port();
```

Comment:
that's not always correct - 

lhs: address = 192.168.0.1, host = "", port = 0
rhs: address = nullptr, host = "", port = ""

As is, the function returns `true` even though it should return `false`. I think that your conditions need to be && with `this->address() == rhs.address()`.

let's add a unit case for this particular case.

Reply:
Added the test case.  It passes because one of the addresses is null but the hostnames are then 
192.168.0.1 and "" respectively.
Honestly we could just always compare pure hostname and skip the address bits for clarity if you'd like?

Reply:
>Added the test case. It passes because one of the addresses is null but the hostnames are then
192.168.0.1 and "" respectively.
Ah, I see - my bad.

>Honestly we could just always compare pure hostname and skip the address bits for clarity if you'd like?
Good idea. Let's do that